pkg/hintrunner/zero/zerohint_test.go

```diff
@@ -0,0 +1,385 @@
+					{name: "a", kind: apRelative, felt: feltAdd(&utils.FeltMax128, feltInt64(1))},
+				},
+				makeHinter: func(ctx *testContext) hinter.Hinter {
+					return newIsNNOutOfRangeHint(ctx.operanders["a"])
+				},
+				check: apValueEquals(feltUint64(1)),
+			},
+
+			{
+				operanders: []*operander{
+					{name: "a", kind: apRelative, felt: feltAdd(&utils.FeltMax128, feltInt64(-1))},
+				},
+				makeHinter: func(ctx *testContext) hinter.Hinter {
+					return newIsNNOutOfRangeHint(ctx.operanders["a"])
+				},
+				check: apValueEquals(feltUint64(0)),
+			},
+		},
+	}
+
+	runTest := func(t *testing.T, tc testCase) {
+		vm := VM.DefaultVirtualMachine()
+		ctx := &hinter.HintRunnerContext{}
+		if tc.vminit != nil {
+			tc.vminit(vm)
+		}
+
+		testCtx := &testContext{
+			vm:          vm,
+			hintContext: ctx,
+			operanders:  make(map[string]hinter.ResOperander),
+		}
+
+		// There are always a few extra values on the memory stack
+		// above FP to make AP-based and FP-based addressing makes more sense.
+		// These elements are identical to their index: mem[0] is 0, mem[1] is 1.
+		//
+		// Since these values are *below* FP, they can be considered to be arguments
+		// to the function; accessed as [fp-1], etc.
+		extraValues := []*fp.Element{
+			feltUint64(0),
+			feltUint64(1),
+			feltUint64(2),
+			feltUint64(3),
+		}
+		for _, v := range extraValues {
+			runnerutil.WriteTo(vm, VM.ExecutionSegment, vm.Context.Ap, memory.MemoryValueFromFieldElement(v))
+			vm.Context.Ap++
+		}
+		// FP points *after* the last extra value.
+		vm.Context.Fp = uint64(len(extraValues))
+
+		for _, o := range tc.operanders {
+			switch o.kind {
+			case apRelative, fpRelative:
+				o.memoryOffset = vm.Context.Ap
+				runnerutil.WriteTo(vm, VM.ExecutionSegment, vm.Context.Ap, memory.MemoryValueFromFieldElement(o.felt))
+				vm.Context.Ap++
+
+			case immediate:
+				// Nothing to do.
+
+			default:
+				panic("unexpected operander kind")
+			}
+		}
+
+		// Now that we filled the memory with values, we can
+		// compute the relative addresses for operanders.
+		for _, o := range tc.operanders {
+			switch o.kind {
+			case apRelative:
+				relOffset := int(vm.Context.Ap - o.memoryOffset)
+				testCtx.operanders[o.name] = &hinter.Deref{
+					Deref: hinter.ApCellRef(-relOffset),
+				}
+
+			case fpRelative:
+				relOffset := int(vm.Context.Fp - o.memoryOffset)
+				testCtx.operanders[o.name] = &hinter.Deref{
+					Deref: hinter.FpCellRef(relOffset),
+				}
+
+			case immediate:
+				testCtx.operanders[o.name] = hinter.Immediate(*o.felt)
+			}
+		}
+
+		h := tc.makeHinter(testCtx)
+
+		err := h.Execute(vm, ctx)
+		if tc.errCheck == nil {
+			// By default, an error should be nil.
+			require.Nil(t, err)
+		} else {
+			tc.errCheck(t, testCtx, err)
+		}
+
+		if tc.check == nil {
+			// Allow omitting a VM state check if errCheck is present.
+			if tc.errCheck == nil {
+				t.Fatal("check and errCheck can't both be nil")
+			}
+		} else {
+			tc.check(t, testCtx)
+		}
+	}
```

Comment:
I don't know yet if there is any merit in doing both, but this code runs both functions if they're present.
The `both nil` test is a sanity check.
It's possible to only allow one of the two and make the code a little simpler.

Reply:
It doesn't do any harm to leave it as it is now. I was just asking in case there was no point on running both of them, but maybe it could be of interest in some case